R/otp-route.R

```diff
@@ -0,0 +1,262 @@
+#' Get the geometry of a route from the OTP
+#'
+#' @param otpcon OTP connection object produced by otp_connect()
+#' @param fromPlace Numeric vector, Latitude/Longitude pair, e.g. `c(51.529258,-0.134649)`
+#' @param toPlace Numeric vector, Latitude/Longitude pair, e.g. `c(51.506383,-0.088780,)`
+#' @param mode Character vector of modes of travel valid values TRANSIT, WALK, BICYCLE, CAR, BUS, RAIL, default CAR
+#' @param date_time POSIXct, a date and time, defaults to current date and time
+#' @param arriveBy Logical, Whether the trip should depart or arrive at the specified date and time, default FALSE
+#' @param maxWalkDistance Numeric passed to OTP
+#' @param walkReluctance Numeric passed to OTP
+#' @param transferPenalty Numeric passed to OTP
+#' @param minTransferTime Numeric passed to OTP
+#' @param full_elevation Logical, should the full elevation profile be returned, default FALSE
+#'
+#' @export
+#'
+#' @details
+#' This function returns a SF data.frame with one row for each leg of the journey
+#' (a leg is defined by a change in mode). For transit more than one route option may be returned
+#' and is indicated by the route_option column.
+#'
+#'
+#' Elevation
+#' OTP supports elevation data, and can return the elevation profile of the route if available.
+#' OTP returns the elevation profile separately from the XY coordinates, this means there is not
+#' direct match between the number of XY points and the number of Z points.  OTP also only returns
+#' the elevation profile for the first leg of the route (this appears to be a bug).
+#' As default the otp_plan function matches the elevation profile to the XY coordinates to return
+#' a SF linestring with XYZ coordinates. If you require a more detailed elevation profile,
+#' the full_elevation parameter will return a nested data.frame with three columns.
+#' first and second are returned from OTP, while distance is the cumulative distance along the
+#' route and is derived from First.
+#'
+otp_plan <- function(otpcon = NA,
+           fromPlace = NA,
+           toPlace = NA,
+           mode = "CAR",
+           date_time = Sys.time(),
+           arriveBy = FALSE,
+           maxWalkDistance = 800,
+           walkReluctance = 2,
+           transferPenalty = 0,
+           minTransferTime = 0,
+           full_elevation = FALSE)
+{
+  # Check Valid Inputs
+  checkmate::assert_class(otpcon,"otpconnect")
+  checkmate::assert_numeric(fromPlace, lower =  -180, upper = 180, len = 2)
+  fromPlace <- paste(fromPlace, collapse = ",")
+  checkmate::assert_numeric(toPlace, lower =  -180, upper = 180, len = 2)
+  toPlace <- paste(toPlace, collapse = ",")
+  mode <- toupper(mode)
+  checkmate::assert_subset(mode, choices = c("TRANSIT","WALK","BICYCLE","CAR","BUS","RAIL"), empty.ok = F)
+  mode <- paste(mode, collapse = ",")
+  checkmate::assert_posixct(date_time)
+  date <- format(date_time, "%m/%d/%Y")
```

Comment:
This is not the correct date format and does not work . Date must be in format:

mm-dd-yyyy
```suggestion
  format(date, "%m-%d-%Y")
```

Reply:
@mem48 @Robinlovelace 
actually it looks like it does work, but think it's safer to be consistent with the web GUI, and avoid passing forward slahes to the api?